cognite/client/data_classes/_base.py

```diff
@@ -140,6 +141,15 @@ def dump(self, camel_case: bool = True) -> dict[str, Any]:
         """
         return basic_instance_dump(self, camel_case=camel_case)
 
+    def dump_yaml(self, filepath: str | Path) -> None:
```

Comment:
I don't find it very intuitive that .dump() returns a dict while .dump_yaml() writes to a file. Let's make it `dump_yaml_to_file()`?

Reply:
Agreed. The expected output from dump_yaml() would be str?

Reply:
Yes I'd say so

Reply:
Agree, and I also think Peter suggestion is a better way, just return the string.